main.js

```diff
@@ -625,81 +699,173 @@ function readValue(id, cb) {
-    }
-    else {
-        cmd = 'set ' + fhemObjects[id].native.Name + ' ' + fhemObjects[id].native.Attribute + ' ' + val;
-    }
-    adapter.log.info(adapter.namespace + '.' + fhemObjects[id].native.Name + '.' + fhemObjects[id].native.Attribute + '.' + val_org + ' ==> writeFHEM: ' + cmd);
-    // edit end LausiD 05.03.17
-
+    } else {
+            cmd = 'set ' + fhemObjects[id].native.Name + ' ' + fhemObjects[id].native.Attribute + ' ' + val;
+            }
+    adapter.log.info('Event ioBroker: ' + id + ' ' + val_org + ' ==> writeFHEM: ' + cmd);
     telnetOut.send(cmd, function (err, result) {
-        if (err) adapter.log.error('writeValue: ' + err);
+        if (err) adapter.log.error('[writeValue] ' + err);
         if (cb) cb();
     });
 }
 
-function requestMeta(name, attr, value, cb) {
-    if (cb) cb();
-    var _id = adapter.namespace + '.' + name.replace(/\./g, '_') + '.' + attr.replace(/\./g, '_');
-    if (fhemObjects[_id]) {
-        parseEvent(name + ' ' + attr + ' ' + value);
-        if (cb) {
-            cb();
-            cb = null;
+//--------------------------------------------------------------------------------------
+function requestMeta(name, attr, value, event, cb) {
+       // send command JsonList2
+        telnetOut.send('jsonlist2 ' + name, function (err, result) {
+        if (err) {
+            adapter.log.error('[requestMeta] ' + err);
         }
-    } else {
-        telnetOut.send('JsonList2 ' + name, function (err, result) {
+        if (result) {
             var objects = null;
             try {
                 objects = JSON.parse(result)
             } catch (e) {
-                adapter.log.error('Cannot parse answer for jsonlist2: ' + e);
+                adapter.log.error('[requestMeta] Cannot parse answer for jsonlist2: ' + e);
             }
             if (objects) {
-                parseObjects(objects.Results, function () {
-                    var id = adapter.namespace + '.' + name.replace(/\./g, '_') + '.' + attr.replace(/\./g, '_');
-                    if (fhemObjects[id]) {
-                        parseEvent(name + ' ' + attr + ' ' + value);
-                    } else {
-                        adapter.log.warn('Readings "' + attr + '" still not found in "' + name + '" after JsonList2');
-                    }
+                    parseObjects(objects.Results, function () {
                     if (cb) {
                         cb();
                         cb = null;
                     }
-                });
+                   });
             } else if (cb) {
                 cb();
                 cb = null;
             }
+        } else if (cb) {
+            cb();
+            cb = null;
+        }
+    });
+}
+
+//--------------------------------------------------------------------------------------
+function deleteChannel(name, cb) {
+
+    adapter.getForeignObject(name, function (err, result) {
```

Comment:
This info belongs to the next methods in general:
"Foreign objects" are normally objects outside of the own adapter namespace (e.g. fhem.0.*). Do you really ned to access outside? If not please use "getObject". But also in general why you not call deleteObject directly but use this "wrapper methods"? ALso to make the "getObject" before an second operation should not be needed